driver/event_table.c

```diff
@@ -63,10 +63,10 @@ const struct ppm_event_info g_event_info[PPM_EVENT_MAX] = {
 	/* PPME_SOCKET_GETPEERNAME_X */{"getpeername", EC_NET, EF_DROP_FALCO, 0},
 	/* PPME_SOCKET_SOCKETPAIR_E */{"socketpair", EC_IPC, EF_CREATES_FD | EF_MODIFIES_STATE, 3, {{"domain", PT_FLAGS32, PF_DEC, socket_families}, {"type", PT_UINT32, PF_DEC}, {"proto", PT_UINT32, PF_DEC} } },
 	/* PPME_SOCKET_SOECKETPAIR_X */{"socketpair", EC_IPC, EF_CREATES_FD | EF_MODIFIES_STATE, 5, {{"res", PT_ERRNO, PF_DEC}, {"fd1", PT_FD, PF_DEC}, {"fd2", PT_FD, PF_DEC}, {"source", PT_UINT64, PF_HEX}, {"peer", PT_UINT64, PF_HEX} } },
-	/* PPME_SOCKET_SETSOCKOPT_E */{"setsockopt", EC_NET, EF_NONE, 0},
-	/* PPME_SOCKET_SETSOCKOPT_X */{"setsockopt", EC_NET, EF_NONE, 0},
-	/* PPME_SOCKET_GETSOCKOPT_E */{"getsockopt", EC_NET, EF_DROP_FALCO, 0},
-	/* PPME_SOCKET_GETSOCKOPT_X */{"getsockopt", EC_NET, EF_DROP_FALCO, 0},
+	/* PPME_SOCKET_SETSOCKOPT_E */{"setsockopt", EC_NET, EF_USES_FD, 4, {{"fd", PT_FD, PF_DEC}, {"level", PT_FLAGS8, PF_DEC, sockopt_levels}, {"optname", PT_FLAGS8, PF_DEC, sockopt_options}, {"val", PT_DYN, PF_DEC, sockopt_dynamic_param, PPM_SOCKOPT_IDX_MAX}}},
+	/* PPME_SOCKET_SETSOCKOPT_X */{"setsockopt", EC_NET, EF_USES_FD, 1, {{"res", PT_ERRNO, PF_DEC}}},
```

Comment:
It seems like we prefer returning information in the exit event, to avoid the overhead of retrieving the enter event in the exit event, where you usually end up doing the parsing. @mattpag would you agree?

Reply:
Yes I agree, most of the time we want to ignore/treat differently failed syscalls, so having everything in the exit makes its handling easier (both wrt performance and implementation).

Reply:
OK, will update.